containers/themes/ThemesSection.js

```diff
@@ -3,45 +3,49 @@ import { c } from 'ttag';
 import {
     useMailSettings,
     useEventManager,
-    useApiWithoutResult,
+    useApi,
+    useLoading,
     SubTitle,
     Alert,
     ThemeCards,
     useModals,
     useNotifications
 } from 'react-components';
 import { updateTheme } from 'proton-shared/lib/api/mailSettings';
-import { getThemeIdentifier } from 'react-components/helpers/themes';
-import { THEMES } from 'proton-shared/lib/constants';
+import { getThemeIdentifier, stripThemeIdentifier } from 'proton-shared/lib/themes/helpers';
+import { ALL_THEMES } from 'proton-shared/lib/themes/themes.js';
 
-import { availableThemes } from './availableThemes.js';
 import CustomThemeModal from './CustomThemeModal.js';
 
-const {
-    CUSTOM: { identifier: customId }
-} = THEMES;
+const { DARK, CUSTOM } = ALL_THEMES;
+const availableThemes = [DARK, CUSTOM];
 
 const ThemesSection = () => {
+    const api = useApi();
     const { createNotification } = useNotifications();
     const { createModal } = useModals();
     const [{ Theme }] = useMailSettings();
     const { call } = useEventManager();
-    const { request, loading } = useApiWithoutResult(updateTheme);
-    const themeId = getThemeIdentifier(Theme);
-    const customCSS = themeId === customId ? Theme : '';
+    const [loading, withLoading] = useLoading();
+    const themeIdentifier = getThemeIdentifier(Theme);
+    const customCSS = themeIdentifier === CUSTOM.identifier ? Theme : '';
 
-    const handleChangeTheme = async (themeId) => {
-        if (themeId === customId) {
+    const themes = availableThemes.map(({ identifier, getI18NLabel, src, customizable }) => {
+        const id = stripThemeIdentifier(identifier);
+        return { identifier, id, label: getI18NLabel(), alt: id, src, customizable };
+    });
+
+    const handleChangeTheme = async (themeIdentifier) => {
+        if (themeIdentifier === CUSTOM.identifier) {
             return handleOpenModal();
         }
-        await request(themeId);
-        call();
+        await api(updateTheme(themeIdentifier));
+        await call();
```

Comment:
Again, can we add a success notification here?

Reply:
Yep, added!